### PR TITLE
Introduce cluster.index.restrict.replication.type setting

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description
Adds documentation for newly introduced setting `cluster.index.restrict.replication.type`.  When enabled this setting supplements `cluster.indices.replication.strategy` setting by ensuring the index level replication type setting `index.replication.type` matches with replication type defined at cluster level i.e. `cluster.indices.replication.strategy`. 

### Issues Resolved
Fixes https://github.com/opensearch-project/documentation-website/issues/5806

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dreamer-89** [comment]
Request for review @hdhalter @Naarcha-AWS

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @dreamer-89! LGTM (made a couple of wording adjustments).

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:90]:
```suggestion
If enabled, the `cluster.index.restrict.replication.type` setting requires newly created indexes to have a replication type specified in the `cluster.indices.replication.strategy` cluster setting. Requests that specify a different replication type are rejected. If `cluster.index restrict.replication.type` is disabled, you can choose a replication type on a per-index basis by specifying it in the `index.replication.type` setting. 
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws One minor change. Otherwise, LGTM!

**Comment by dreamer-89** [comment]
@natebower @kolchfa-aws : Is there anything pending thats needs to be fixed before merge ?

**Comment by natebower** [comment]
> @natebower @kolchfa-aws : Is there anything pending thats needs to be fixed before merge ?

@dreamer-89 Not from me 😄

**Comment by kolchfa-aws** [comment]
@dreamer-89 The PR is ready to merge but we usually merge right before the release. I'll take care of it when the time comes :)

**Pull request merged**
